jena-arq/src/main/java/org/apache/jena/sparql/exec/RowSetOps.java

```diff
@@ -51,7 +51,7 @@ public static void consume(RowSet rowSet)
      * This operation consumes the RowSet.
      */
     public static long count(RowSet rowSet)
-    { return rowSet.rewindable().size(); }
+    { long c[] = {0}; rowSet.forEach(b -> ++c[0]); return c[0]; }
```

Comment:
Intent: Avoid memory copy.

Reply:
Or maybe `rowSet.stream().count()` would be cleaner and still avoid the memory copy (though it has conversion to `Stream` overhead?

Reply:
good point!

Reply:
Maybe it does not matter with thhis distructive operation but the cost of `stream()` is not trivial. 
`forEach` does not use `stream`.

Reply:
> Intent: Avoid memory copy.

`RowSetMem` rewind should not be a memory copy. It shares the rows array.

A rewindable RowSet (e.g. Mem) goes back to the beginning where as `forEach` continues the iteration.

Reply:
Behavior so far:
* For non-RowSetMem instances counting would create an intermediate in-memory copy.
* The count for a RowSetMem instances would be the number of bindings in the row set - rather than the remaining bindings.

I added a the method `RowSetStream.forEachRemaining` - the stack trace for `RowSetOps.count` then becomes a delegate to the forEachRemaining method (with some boilerplate):

![image](https://github.com/user-attachments/assets/a9726dfa-4a7b-4dbc-b243-47ef2a18db43)